### ISSUE TITLE
Update to Gradle 5.6.x

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#1520]
we should update to Update to Gradle 5.6.x once its out

https://github.com/eclipse/xtext/blob/master/dev-doc/GradleWrapperVersionBump.md

https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20663

**Comment by cdietrich** [comment]
```
Java Model Exception: Java Model Status [Test source folder 'testdata' in project 'org.eclipse.xtext.xbase.testing' must have an output folder that is not also used for main sources]
	at org.eclipse.jdt.internal.core.JavaModelOperation.runOperation(JavaModelOperation.java:798)
	at org.eclipse.jdt.internal.core.JavaProject.setRawClasspath(JavaProject.java:3554)
	at org.eclipse.jdt.internal.core.JavaProject.setRawClasspath(JavaProject.java:3514)
	at org.eclipse.jdt.internal.core.JavaProject.setRawClasspath(JavaProject.java:3570)
	a
```

**Comment by cdietrich** [comment]
Description	Resource	Path	Location	Type
Failed to configure project org.eclipse.xtend.core.tests	gradle-composite		line 0	Gradle Error Marker
Failed to configure project org.eclipse.xtext.common.types.tests	gradle-composite		line 0	Gradle Error Marker
Failed to configure project org.eclipse.xtext.ide.tests	gradle-composite		line 0	Gradle Error Marker
Failed to configure project org.eclipse.xtext.xbase.testing	gradle-composite		line 0	Gradle Error Marker

**Comment by cdietrich** [comment]
created https://github.com/eclipse/buildship/issues/909

**Comment by cdietrich** [comment]
Potential Workaround

```
if (source.kind == 'src') {
					if (project.name.endsWith("tests")) {
						source.entryAttributes['test'] = 'true'
					} else {
						source.entryAttributes['test'] = 'false'
					}
				}
```

have still no clue why buildship/gradle does what it does
cc @szarnekow 

do you have any idea @oehme

**Comment by cdietrich** [comment]
also projects created with with the wizard and plan layout dont properly mark source folders in the .tests project as test

org.xtext.example.mydsl1.tests

**Comment by cdietrich** [comment]
i would prefer not having to rely on the workaround. thus i prepared a patch for gradle. but that indicates we have to update to a later gradle version and not 5.6

**Comment by cdietrich** [comment]
my patch is now in the nightlys. it solves most of the problems, but there is still issues with custom sourcesets.

**Comment by cdietrich** [comment]
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20761

**Comment by cdietrich** [comment]
Fixed in 2.20
